src/bn256/fq2.rs

```diff
@@ -9,7 +9,7 @@ use std::cmp::Ordering;
 use subtle::{Choice, ConditionallySelectable, ConstantTimeEq, CtOption};
 
 /// An element of Fq2, represented by c0 + c1 * u.
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
I figured it didn't hurt, just in case we want a HashMap with `G2Affine` for some reason in the future?

Reply:
I'd not add it for now if it's not needed for a particular feature. I don't see the usecase for having `G2` elements inside of a HashMap neither. You'll also be able to compress the G2 elem and `Hash` is implemented for `[u8;N]` so you're still able to do it.

Happy to include this if we have a clear use-case in the future!!

Reply:
Ah if I turn it off then the macro can't derive hash for `G1Affine` either since they both use `new_curve_impl`. I do think it's useful for `G1Affine` because for example in `plonk-verifier` I think some caching currently uses `BTreeMap` but `FxHashMap` would be faster. I guess I can implement it manually as you say; it's just a little cumbersome from the dev perspective.